src/main/java/com/example/airbnbapi/service/MediaService.java

```diff
@@ -91,6 +94,109 @@ public boolean itemExists(MediaType type, String title) {
         }
         return false;
     }
+
+    //// authentication ////
+
+    // add an authenticated user to the authenticated db
+    public OauthUser authenticateUser(User user) {
+
+        if (authenticatedUserExists(user.getEmailAddress())) {
+            return null;
+        }
+
+
+        try {
+
+            String hashedPassword = BCrypt.hashpw(user.getPassword(), BCrypt.gensalt());
+
+            OauthUser oauthUser = new OauthUser();
+            oauthUser.setOauthCode(UUID.randomUUID());
+            oauthUser.setEmailAddress(user.getEmailAddress());
+            oauthUser.setPassword(hashedPassword);
+
+            return oauthUserRepository.save(oauthUser);
+        } catch (RuntimeException e) {
+
+            logger.error("An error occurred when trying to request data from the database, RuntimeException: " + e);
+            throw new DataBaseException("Sorry, couldn't establish connection to the database", e);
+        }
+
+    }
+
+    // adds a user to the user db.
+    public User addUser(User user) {
+
+        List<User> allUsers = userRepository.findAll();
+
+        try {
+            for (int i = 0; i <= allUsers.size(); i++) {
+
+                if (allUsers.size() == 0 || !allUsers.get(i).getEmailAddress().equals(user.getEmailAddress())) {
+
+                    return userRepository.save(user);
+                }
+            }
+        } catch (RuntimeException e) {
+
+            logger.error("An error occurred when trying to request data from the database, RuntimeException: " + e);
+            throw new DataBaseException("Sorry, couldn't establish connection to the database", e);
+        }
+        return null;
+    }
+
+    // checks that users email and password match what is stored in the user db
+    public boolean userExistsAndPasswordMatches(String emailAddress, String password) {
+        List<User> allUsers = userRepository.findAll();
```

Comment:
Again, perhaps introduce a custom method into your repository to pull back any records with a given email address to avoid needing to loop over all users?